### PR TITLE
[dev] Merge 1.0 branch May 2021 update changes into dev branch

### PR DESCRIPTION
**Pull request by thcrain-msft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

This is a port of the changes from the 1.0 May 2021 Update to the 2.0 development branch. Please see the specific commits for further information.

###### Change Log  

- See the [May 2021 release notes](https://github.com/microsoft/CBL-Mariner/releases/tag/1.0.20210604-1.0) for further information on relevant changes.

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Successful full build

### CODE REVIEW

**Review by danmihai1 — APPROVED** [review]

File: SPECS/bind/bind.spec [hunk 1]
```diff
@@ -412,7 +412,7 @@ fi;
 %dir /run/named
 
 %files libs
-%{_libdir}/*so.*
+%{_libdir}/*-%{version}*.so
```
Comment (PawelWMS) [inline SPECS/bind/bind.spec:415]: The `*.so` files are more common to the `*-devel` subpackages. What happened, that we moved these around? I'm not sure, if this is correct.

**Review by PawelWMS — CHANGES_REQUESTED** [review]

File: SPECS/WALinuxAgent/WALinuxAgent.spec [hunk 2]
```diff
@@ -74,10 +74,12 @@ python3 setup.py check && python3 setup.py test
 %config %{_sysconfdir}/waagent.conf
 %ghost %{_localstatedir}/log/waagent.log
 %dir %attr(0700, root, root) %{_sharedstatedir}/waagent
-%{_lib}/python3.7/site-packages/*
+%{python3_sitelib}/*
 
 %changelog
 * Mon Jun 21 2021 Thomas Crain <thcrain@microsoft.com> - 2.2.54.2-1
```
Comment (PawelWMS) [inline SPECS/WALinuxAgent/WALinuxAgent.spec:80]: We shouldn't be changing the history for a spec. The original from `1.0-dev` with this version and release number was authored by Suresh and didn't include the lib macro updates. I'd revert to the original changelog entry and just add another release for `dev` on top of it.

File: SPECS/bind/bind.spec [hunk 3]
```diff
@@ -533,6 +533,9 @@ fi;
 
 %changelog
 * Mon Jun 21 2021 Thomas Crain <thcrain@microsoft.com> - 9.16.15-1
```
Comment (PawelWMS) [inline SPECS/bind/bind.spec:535]: The `1.0-dev` changelog entry for this version and release was done by Andrew. If you need to apply additional changes to this spec to make it work for `dev`, please keep the original version from `1.0-dev` and apply any fixes as a separate release.

**Review by PawelWMS — CHANGES_REQUESTED** [review]

File: SPECS/bind/bind.spec (hunk 1 above)
Comment (thcrain-msft) [reply SPECS/bind/bind.spec:415]: [As of bind 9.16.12](http://gitlab.isc.org/isc-projects/bind9/-/blob/v9_16/CHANGES#L372), the installed libraries have had their sonames changed to include the version number. This versioning replaces the versioning after the `.so`. I believe this is to discourage third party applications from using the libraries.

[Fedora also packages the libraries in this way.](https://src.fedoraproject.org/rpms/bind/blob/rawhide/f/bind.spec#_963)

**Pull request closed without merging**